README.md

```diff
@@ -11,47 +11,62 @@
-i.e. one with a C-like `main`, in a minimal seL4 environment.
+This repository provides a minimal runtime for running a C or C-compatible process, 
+i.e., one with a C-like `main()`, in a minimal seL4 environment.
 
 This runtime provides mechanisms for accessing everything a standard
-process would expect to need at start and provides additional utilities
-for delegating the creation of processes and threads.
+process or thread would expect to need at start, including thread
+local storage.
 
-## Standard Processes
+To use the library, you just need to write a C program that starts
+from `main()`, and compile it into an ELF file; the documentation
+below gives implementation details.  There is a CMakeLists.txt file
+provided to hook sel4runtime into your project.
 
-All processes (except for the root task) will use the entry-points
-provided here as normal and require the `_start` entry-point provided in
+## Implementation
+
+### Root Task
+
+The root task needs some extra work before `main()` is called.  The
+runtime calls `_sel4_start()`, provided by this library, that takes
+`seL4_BootInfo` as argument in a register, and has not been given a
+stack.
+
+This entry-point sets up a static 16 kilobyte stack before invoking
+`__sel4_start_root()`, also provided by the runtime, which
+constructs the argument, environment, and auxiliary vectors. It then
+passes the constructed vectors, along with the address of `main()`,
+into `__sel4_start_main` which configures the runtime according to the
+architecture's conventions before starting calling `main()`.
+
+### Standard Processes
+
+All processes other than the root task use the `_start` entry-point provided in
 the architecture-dependant `crt0.S`. This will then bootstrap into the
 runtime entry-point `__sel4_start_c` which simply processes the stack to
 find the argument, environment, and auxiliary vectors.
 
-The found vectors, along with`main`, are passed into
+These vectors, along with the address of `main()`, are passed into
 `__sel4_start_main` which configures the runtime before starting
-`main`.
+`main`.  The architecture's standard post-main() processing is handled
+by `__start`.
 
-## Root Task
 
-The root task requires an alternate entry-point `_sel4_start` which
-assumes that the `seL4_BootInfo` argument has been passed to it and that
-it has not been given a stack.
+### Thread-local storage layout
 
-This entry-point moves onto a static 16 kilobyte stack before invoking
-`__sel4_start_root`, which constructs the argument, environment, and
-auxiliary vectors. It then passes the constructed vectors, along with
-`main`, into `__sel4_start_main` which configures the runtime before
-starting `main`.
+There are two supported layouts for thread local storage.
 
-## Thread-local storage layout
+Intel X86\_64 and IA32 use a TLS pointer that contains the address
+immediately after the TLS region: the region is _below_ it in memory.
```

Comment:
I'd personally prefer _before_ and _after_ in place of _above_ and _below_ as I personally know a non-insignificant amount of people with either possible interpretation of what it would mean for on memory region to be _above_ another.

Reply:
Above --> higher addresses; below -> lower addresses.  I'll clarify.

Reply:
I don't think that there is any consistency with people's understanding of the terms below or above and would prefer to avoid their use altogether.